app/mails/tools.py

```diff
@@ -24,9 +25,24 @@ def get_delay_days_from_email_address(email_address):
     :rtype: integer
     """
     try:
-        match = re.findall(r"^(\d+)([dmw])", email_address)[0]
-        multiplicator = settings.EMAIL_SUFFIX_TO_DAY[match[1]]
-        delay = int(match[0]) * int(multiplicator)
+        if re.match(r"^(\d+)([dmw])", email_address):
+            match = re.findall(r"^(\d+)([dmw])", email_address)[0]
+            multiplicator = settings.EMAIL_SUFFIX_TO_DAY[match[1]]
+            delay = int(match[0]) * int(multiplicator)
+        elif re.match(r"(\.[0-9a-z]{10})@", email_address):
+            key = re.search(r"(\.[0-9a-z]{10})@", email_address).group(1)
+            email_without_key = email_address.replace(key, "")
+            delay_datetime = dateparser.parse(
+                email_without_key.split("@")[0], settings=settings.DATEPARSER_SETTINGS
+            )
+            delay = (delay_datetime.date() - timezone.now().date()).days
+        else:
+            delay_datetime = dateparser.parse(
+                email_address.split("@")[0], settings=settings.DATEPARSER_SETTINGS
+            )
+            delay = (delay_datetime.date() - timezone.now().date()).days
+        if delay < 0:
+            raise Exception("Invalid delay")
         return delay
```

Comment:
Is there a better way to solve this? Raising two exceptions seems kinda unnecessary